plugins/experimental/ssl_session_reuse/src/ssl_key_utils.cc

```diff
@@ -368,6 +372,10 @@ STEK_update(const std::string &encrypted_stek)
 static void *
 STEK_Update_Checker_Thread(void *arg)
 {
+  plugin_threads.store(::pthread_self());
+  ::pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, nullptr);
+  ::pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, nullptr);
```

Comment:
how is this a good idea?

This leads to resource leaks as there is no way for the thread to respond to the "kill" of the thread. Should we not provide a message to the thread for it to read and shutdown with?

Reply:
I talked to the duke.. I understand this is a temp fix to a larger issue to address